src/Api/Response/ResponseParser.php

```diff
@@ -3,228 +3,160 @@
     {
-        $parsedXml = simplexml_load_string($xmlString);
-
-        if (!$parsedXml) {
-            throw new Exception('Error while parsing XML');
-        }
-
-        return $parsedXml;
+        return $this->response->getResult()->getMetadata()->getLandingPage();
     }
 
-    public function createResponseObject(): Response
+    public function getPromotionExtension(): ?Promotion
     {
-        return pluginApp(Response::class);
+        return $this->response->getResult()->getMetadata()->getPromotion();
     }
 
-    /**
-     * @param SimpleXMLElement $data
-     * @return array
-     */
-    protected function parseServers(SimpleXMLElement $data)
+    public function parseTotalResults() :int
     {
-        $servers = [];
-
-        if (!empty($data->servers)) {
-            $servers['frontend'] = $data->servers->frontend->__toString();
-            $servers['backend'] = $data->servers->backend->__toString();
-        }
-
-        return $servers;
+        return $this->response->getResult()->getMetadata()->getTotalResults() ?: 0;
     }
 
-    /**
-     * @param SimpleXMLElement $data
-     * @return array
-     */
-    protected function parseQuery(SimpleXMLElement $data)
+    public function getProductIds() :array
     {
-        $query = [];
-
-        if (!empty($data->query)) {
-            $query['query'] = $data->query->queryString->__toString();
-            $query['searchedWordCount'] = $data->query->searchWordCount->__toString();
-            $query['foundWordCount'] = $data->query->foundWordCount->__toString();
-
-            $query['first'] = $data->query->limit['first']->__toString();
-            $query['count'] = $data->query->limit['count']->__toString();
-        }
-
-        return $query;
+        return array_map(
+            function (Item $product) {
+                if (count($product->getVariants())) {
+                    return  $product->getVariants()[0]->getId();
+                } 
+                else if(array_key_exists('variation_id', $product->getProperties())){
+                    return $product->getProperties()['variation_id'];
+                }
+                else {
+                    return $product->getId();
+                }
+            },
+            $this->response->getResult()->getItems()
+        );
     }
 
-    /**
-     * @param SimpleXMLElement $data
-     * @return string|null
-     */
-    protected function parseLandingPage(SimpleXMLElement $data)
+    public function getFiltersExtension(): FiltersExtension
     {
-        if (!isset($data->landingPage)
-            || empty($data->landingPage->attributes())
-            || !isset($data->landingPage->attributes()->link)
-        ) {
-            return null;
+        $mainFilters = $this->response->getResult()->getMainFilters();
+        $otherFilters = $this->response->getResult()->getOtherFilters();
+        $filtersExtension = pluginApp(FiltersExtension::class);
```

Comment:
1. This classes within the Struct folder are also something more Shopware specific. We do not really work with structs here. You can simply move them to some sub folder here.
2. Can't we simply use `new FilterExtension()`?

Same applies for the other Struct classes